src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -146,7 +152,8 @@ public void testFromXContent() throws IOException {
     @Test
     public void testToQuery() throws IOException {
         QueryParseContext context = new QueryParseContext(index, queryParserService);
-        assertLuceneQuery(this.testQuery, this.testQuery.toQuery(context));
+        context.setAllowUnmappedFields(true);
```

Comment:
For the TermQueryBuilder that is tested in this PR there are two paths in `toQuery(context)` depending on whether `smartFieldMappers` are found or not in the context. I think in this case we don't need to test all possible cases since it's hard to test all the internals of the produced lucene query without getting to implementation specific (doing casts, instanceof etc...) so I would opt for using `unmappedFieldAsString` for simplicity here. 
However, when we see writing tests for other queries that we need to test different execution paths depending on whether there is a mapping or not, we can fake other QueryParseContexts there and overwrite the BaseQueryTestCase#testToQuery() case there.